packages/core/src/service/Project.ts

```diff
@@ -390,17 +389,22 @@ export class Project implements ExternalEventEmitter {
 	private setInitPromise(): void {
 		const loadConfig = async () => {
 			this.config = await this.#configService.load()
-		}
-		const loadIgnore = async () => {
-			const uri = this.projectRoot + Project.GitIgnore
-			try {
-				const gitignore = bufferToString(await this.externals.fs.readFile(uri))
-				this.#ignore = Project.DefaultIgnore.add(gitignore)
-			} catch (e) {
-				if (this.externals.error.isKind(e, 'ENOENT')) {
-					return
+			this.ignore = ignore()
+			for (const pattern of this.config.env.exclude) {
+				if (pattern === '@gitignore') {
+					try {
+						const uri = this.projectRoot + Project.GitIgnore
+						const contents = bufferToString(await this.externals.fs.readFile(uri))
+						this.ignore.add(contents)
+					} catch (e) {
+						if (this.externals.error.isKind(e, 'ENOENT')) {
+							continue
+						}
+						this.logger.error(`[Project] [Config] Failed reading .gitignore`, e)
+					}
```

Comment:
this seems like a good candidate to be refactored into a separate method for readability

`private parseGitIgnore()` maybe

Reply:
Done!